system-helper/flatpak-system-helper.c

```diff
@@ -1225,26 +1225,31 @@ flatpak_authorize_method_handler (GDBusInterfaceSkeleton *interface,
       g_variant_get_child (parameters, 2, "&s", &ref);
       g_variant_get_child (parameters, 3, "&s", &origin);
 
-      is_update = (flags & FLATPAK_HELPER_DEPLOY_FLAGS_UPDATE) != 0;
-      is_app = g_str_has_prefix (ref, "app/");
-
-      if (is_update)
-        {
-          if (is_app)
-            action = "org.freedesktop.Flatpak.app-update";
-          else
-            action = "org.freedesktop.Flatpak.runtime-update";
-        }
+      if (ref != NULL && g_strcmp0 (ref, OSTREE_REPO_METADATA_REF) == 0)
```

Comment:
This could do with a comment justifying why `ostree-metadata` gets special treatment (i.e. the description from the PR).

Reply:
an alternative would be to make this a separate systemhelper method, to make it more obvious, say, UpdateMetadata